From b50258edfda6371dbb5ee8425755f5349098ea0c Mon Sep 17 00:00:00 2001
From: Stephen McConnel <stephen_mcconnel@sil.org>
Date: Thu, 10 May 2012 14:53:06 -0500
Subject: [PATCH] Fix TreeNode removal bug.

This was reported as https://bugzilla.xamarin.com/show_bug.cgi?id=5010
The impetus was FieldWorks bug FWNX-872 "Crash on 'configure dictionary' dailog
box When I remove duplicate Allomorph field".
---
 .../System.Windows.Forms/TreeNodeCollection.cs     |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

Index: fieldworks-mono/mcs/class/Managed.Windows.Forms/System.Windows.Forms/TreeNodeCollection.cs
===================================================================
--- fieldworks-mono.orig/mcs/class/Managed.Windows.Forms/System.Windows.Forms/TreeNodeCollection.cs
+++ fieldworks-mono/mcs/class/Managed.Windows.Forms/System.Windows.Forms/TreeNodeCollection.cs
@@ -375,6 +375,8 @@
 				tree_view.RecalculateVisibleOrder (prev);
 
 				if (removed == tree_view.SelectedNode) {
+					if (removed.IsExpanded)
+						removed.Collapse();	// Fix Xamarin Bugzilla 5010.
 					re_set_selected = true;
 					OpenTreeNodeEnumerator oe = new OpenTreeNodeEnumerator (removed);
 					if (oe.MoveNext () && oe.MoveNext ()) {
